coalib/coala.py

```diff
@@ -59,11 +60,34 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
+                get_all_bears)
+            filtered_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead.")
+                args.filter_by = [] if args.filter_by is None else \
+                    args.filter_by
+                args.filter_by.append(['language'] + args.filter_by_language)
```

Comment:
If we use `extend`, it would be:
`['language', 'python']` and not `[['language', 'python']]`. `extend` essentially appends all the elements present in the list given as argument. But, we want to append it as list and not as individual elements.

Regarding, `+=`, it actually creates new object. Since, `args.filter_by` is not our own variable, and its created by `argparse`, its not advised to change its reference. It might to lead to unexpected behavior. And there's no advantage of it.